server/utils/file/csvParser.js

```diff
@@ -3,91 +3,112 @@ const R = require('ramda')
 
 class CSVParser {
 
-  constructor (filePathOrStream, readHeaders = true) {
-    this.destroyed = false
-    this.csvStreamEnded = false
-    this.filePathOrStream = filePathOrStream
-    this.readHeaders = readHeaders
-    this.rowReadyListener = null
-    this.error = null
-    this.initialized = false
+  constructor (filePathOrStream) {
+    this._csvStream = null
+    this._csvStreamEnded = false
+    this._filePathOrStream = filePathOrStream
+    this._rowReadyListener = null
+    this._error = null
+    this._initialized = false
+    this._destroyed = false
+    this._headers = null
   }
 
-  init () {
-    this.csvStream = this._createCsvStream({ headers: this.readHeaders })
+  async init () {
+    this._csvStream = this._createCsvStream({ headers: false })
 
-    this.csvStream
+    this._csvStream
       .on('data', data => this._onData(data))
       .on('end', () => this._onStreamEnd())
-      .on('error', error => this.error = error)
-      .pause()
+      .on('error', error => this._error = error)
 
-    this.initialized = true
+    this._initialized = true
   }
 
   calculateSize () {
     return new Promise((resolve, reject) => {
-      if (this.error)
-        reject(this.error)
+      if (this._error)
+        reject(this._error)
 
-      let count = 0
+      let count = -1 //do not include headers
       // do not consume instance csv stream, create a new one
-      this._createCsvStream({ headers: true })
+      this._createCsvStream({ headers: false })
         .on('data', () => count++)
         .on('end', () => resolve(count))
         .on('error', reject)
     })
   }
 
   async next () {
-    if (!this.initialized)
+    if (!this._initialized)
       throw new Error(`${this.constructor.name} not initialized yet`)
 
     return new Promise((resolve, reject) => {
-      if (this.error)
-        reject(this.error)
+      if (this._error)
+        reject(this._error)
 
-      if (this.csvStreamEnded)
+      if (this._csvStreamEnded)
         resolve(null)
 
-      this.rowReadyListener = row => {
-        this.rowReadyListener = null
+      this._rowReadyListener = row => {
+        this._rowReadyListener = null
         resolve(row)
       }
-      this.csvStream.resume()
+      this._csvStream.resume()
     })
   }
 
   destroy () {
-    if (!this.destroyed && this.csvStream) {
-      this.csvStream.destroy()
-      this.csvStream = null
-      this.filePath = null
-      this.rowReadyListener = null
-      this.destroyed = true
+    if (!this._destroyed && this._csvStream) {
+      this._csvStream.destroy()
+      this._csvStream = null
+      this._filePathOrStream = null
+      this._rowReadyListener = null
+      this._headers = null
+      this._destroyed = true
+      this._initialized = false
```

Comment:
there's a lot in common with the constructor variables initialization. can you create a method?

Reply:
good